### ISSUE TITLE
Intel Tiger Lake laptop problem with KDE Neon and elementary OS linux ISOs

### ISSUE THREAD

**Issue by vjr** [ventoy/Ventoy#815]
Hello!

I have a new Intel Tiger Lake based laptop HP Pavilion 14 dv0054tu model with Core i5 1135G7 CPU with Iris Xe iGPU and the following 2 linux ISOs (KDE Neon and elementary OS 6.0 beta) - I get their GRUB boot menu but after selecting to boot it just gives black screen hang.

Note that the same ISOs work on other hardware (Ryzen desktop and older Intel skylake laptop) so can you please let me know what troubleshooting steps I can perform and post the results here? (Since you may not be able to test without similar hardware).

The 2 ISOs are:

https://files.kde.org/neon/images/user/current/neon-user-current.iso

and

https://drive.google.com/file/d/1LGWb_4f3N2S5YMYuuFev_Lkm9mQaeEhR/view?usp=sharing

Thanks!

**Comment by ventoy** [comment]
Does the latest 1.0.39 release have some difference?

**Comment by vjr** [comment]
Hello!

No, the latest release has same issue. Note this works on other hardware (Ryzen desktop and Skylake laptop) and also these 2 ISO if I write them directly to USB stick using Rufus tool then they work on this Tiger Lake laptop, FYI.

Is there any steps I can take to troubleshoot and provide any further info/logs etc?

Thanks!

**Comment by vjr** [comment]
Here are some output from linux OS if it helps?

https://github.com/elementary/os/files/6215033/dmesg.txt

https://github.com/elementary/os/files/6215034/lshw.txt

**Comment by vjr** [comment]
Hello @ventoy  is there anything I can do to help identify the problem whether it's on iso side or ventoy side?

**Comment by steve6375** [comment]
If it works on other systems then it is a hardware\driver issue between linux and your hardware.
Typically you can add kernel parameters to the   linux kernel  command line to overcome the issue such as
acpi=off noapic nolapic edd=on nodmraid nomodeset
You must consult the documentation on each Linux ISO as they are all different!

**Comment by vjr** [comment]
> If it works on other systems then it is a hardware\driver issue between linux and your hardware.
> Typically you can add kernel parameters to the linux kernel command line to overcome the issue such as
> acpi=off noapic nolapic edd=on nodmraid nomodeset
> You must consult the documentation on each Linux ISO as they are all different!

Note that the iso works if I write it directly to the usb stick.

**Comment by steve6375** [comment]
Are you legacy booting in all cases?

**Comment by vjr** [comment]
No, uefi in both cases... anything of interest to look into here?

**Comment by steve6375** [comment]
So when you uefi boot and see a black screen, how do you know it hangs? Does all usb led activity stop immediately? Can you ctrl-alt-del? Could video be directed to a different monitor output? Did you try first switching to text mode in ventoy before running iso?

**Comment by vjr** [comment]
Yes, no response from keyboard,  only pressing power button shuts off the laptop. 

I have not tried text mode,  will try now and report back. 

Thanks for your inputs btw.

**Comment by vjr** [comment]
I tried text mode with the same result. 

I might not have mentioned this before but I do get the grub boot screen where it shows the os logo with countdown to boot.  Upon pressing enter is where it goes blank.

**Comment by steve6375** [comment]
So at menu with countdown, none of the menu options boot?
If you press e you should be able to edit the grub2 Linux kernel line and add in the suggested parameters, did you try?

**Comment by vjr** [comment]
I have tried many options including nomodeset. Not tried edd or nodmraid. Will try it tomorrow though. Thanks for your help again.

**Comment by steve6375** [comment]
And remove any quiet or nosplash options.
Maybe add debug parameter?

**Comment by steve6375** [comment]
Might be interesting to test same ISOs using easy2boot with agFM too.

**Comment by vjr** [comment]
> 
> 
> And remove any quiet or nosplash options.
> Maybe add debug parameter?

Yup tried some more options like `earlyprintk` etc but no luck, it seems to hang early upon boot.

**Comment by steve6375** [comment]
If you remove the  **quiet** **splash**  words in the KDE neon  grub2 menu, add **debug** in its place (**before** the --- at the end of the line of course), then press F10 to boot - what do you see? No text at all after pressing F10 - even if you have Ventoy in text mode first? When you press F10 to boot, does the LED flash on the USB drive?

**Comment by ventoy** [comment]
@vjr 
Please do the test as follows:
1. change to text mode
2. when boot to Ventoy's boot menu, press `c` to enter grub shell and run  `vtdebug on` cmd and then press `ESC` to return.
3. Select the iso to boot.

It may pause many times, just follow the prompt and press Enter to continue.

**Comment by vjr** [comment]
Thank you @ventoy  I tried this and mostly initially steps all seem to show success and again I reach operating system grub menu. Here again I remove the quiet splash options and add nomodeset options then after proceeding it hangs with following ventoy log on screen nothing else...

![20210409_181554](https://user-images.githubusercontent.com/612302/114181880-b0145f00-995f-11eb-86e3-64ec75bccc5a.jpg)

I don't know if this is useful log... is there anyway to save entire log to disk so that I can attach it here?

Does my picture show any useful info to you?

**Comment by vjr** [comment]
Previous picture was from elementary os, the following picture is from kde neon... same hang with similar log...

![20210409_182729](https://user-images.githubusercontent.com/612302/114183387-44cb8c80-9961-11eb-920c-c08f36da974d.jpg)

**Comment by ventoy** [comment]
Besides, Can any other ISO files boot fine with Ventoy in this laptop? (e.g. Ubuntu/CentOS ...)

**Comment by ventoy** [comment]
You can get the vmlinuz file from the ISO and put it in the Ventoy USB stick.
After boot into Ventoy's menu in text mode, press `c` to enter grub shell
and run
```
linux  $vtoy_iso_part/vmlinuz loglevel=10
boot
```

**Comment by vjr** [comment]
> Besides, Can any other ISO files boot fine with Ventoy in this laptop? (e.g. Ubuntu/CentOS ...)

Yes other iso like windows  , fedora etc works...

**Comment by vjr** [comment]
> You can get the vmlinuz file from the ISO and put it in the Ventoy USB stick.
> After boot into Ventoy's menu in text mode, press `c` to enter grub shell
> and run
> 
> ```
> linux  $vtoy_iso_part/vmlinuz loglevel=10
> boot
> ```

Following is the result...

![20210409_195324](https://user-images.githubusercontent.com/612302/114194578-469b4d00-996d-11eb-95a6-efa255a51a1a.jpg)

Anything further to try?

**Comment by ventoy** [comment]
Last try.
press ctrl+i and then select the ISO file and boot.

**Comment by vjr** [comment]
Hello @ventoy  thanks for you help - even the last step did not work - got the same hang. Are you able to download the ISO from the google drive link I posted in an earlier comment in this ticket? Maybe you can look at the contents of the ISO file yourself if you have the time. But it is not an urgent or critical issue for me, I think people will just need to use separate USB stick if they face same issue. Thanks for your help again! Cheers.

**Comment by ventoy** [comment]
Please make a test with this CI release:
https://github.com/ventoy/Ventoy/actions/runs/735293998

Press `ctrl+r` in Ventoy's boot menu and then select the ISO file and boot.

ps: when press ctrl+r there will be red words `GRUB2 mode` in the bottom of the menu screen.

**Comment by vjr** [comment]
Hello @ventoy  !!! This time - SUCCESS ! Both ISOs of KDE Neon and elementary OS 6 (early access) are booting OK with this CI release. Looking forward to the public release whenever it comes. Regards.

**Comment by ventoy** [comment]
This is just a workaround, I still haven't found the cause of the problem.
However, I will add this workaround in the next release.
But it is not enabled by default. So you need to press `ctrl+r`  everytime you boot the two special ISO files.

**Comment by vjr** [comment]
Ok I see. For time being this workaround is very helpful. 

Are you able to download the isos from the links I provided in earlier comment to examine yourself?

If not then I can send you any files from inside the iso of smaller size to examine or if you want I can follow further instructions you may have to investigate further.

**Comment by ventoy** [comment]
I can download the iso files, but they boot fine in my machine :(

**Comment by vjr** [comment]
Just an update that same issue occurs with beta 21.04 release of Ubuntu and Ubuntu budgie isos. 

I will try to report issue to Ubuntu bug tracker and see if they can help identify the root cause.

**Comment by ValdikSS** [comment]
It may be something with memory alignment. This is most likely a buggy UEFI. @ventoy, make sure that Ventoy, when loading .efi file into memory, allocates the memory aligned to 64 pages address (256 KiB). Some UEFI implementations have issues booting file when the memory is not (fully) aligned.

**Comment by vjr** [comment]
@ValdikSS  thanks for you input! good point about UEFI being buggy or at least picky about memory alignment. Just FYI that I had already updated to the latest available BIOS of my laptop and I will update whenever newer versions become available and report back here. But in the mean time if @ventoy  wants to try your suggestion I can gladly test it.

**Comment by ventoy** [comment]
> It may be something with memory alignment. This is most likely a buggy UEFI. @ventoy, make sure that Ventoy, when loading .efi file into memory, allocates the memory aligned to 64 pages address (256 KiB). Some UEFI implementations have issues booting file when the memory is not (fully) aligned.

Ventoy use `LoadImage` and `StartImage` interface of BootService to boot the .efi file.

**Comment by ValdikSS** [comment]
@ventoy, I understand, but some UEFI are buggy and can't load image with `LoadImage`/`StartImage` unless `*SourceBuffer` is aligned. I have not checked, but assume that Ventoy reads BOOTX64.EFI from the ISO file into `malloc`'ed memory and then call `LoadImage` with the pointer to that address. Make sure that pointer is aligned to 64 pages (`%262144`).

**Comment by ventoy** [comment]
> LoadImage

I remember that `LoadImage` only return a handle and doesn't need to provide a Buffer for the image. I will check it.

**Comment by ventoy** [comment]
Besides, only the two iso files have this problem, other iso files boot fine in the same laptop. So it doesn't seem to be a problem with the UEFI firmware. On 04/14/2021 22:57, ValdikSS wrote: 
@ventoy, I understand, but some UEFI are buggy and can't load image with LoadImage/StartImage unless *SourceBuffer is aligned. I have not checked, but assume that Ventoy reads BOOTX64.EFI from the ISO file into malloc'ed memory and then call LoadImage with the pointer to that address. Make sure that pointer is aligned to 64 pages 
[
{
***@***.***": "http://schema.org",
***@***.***": "EmailMessage",
"potentialAction": {
***@***.***": "ViewAction",
"target": "https://github.com/ventoy/Ventoy/issues/815#issuecomment-819585071",
"url": "https://github.com/ventoy/Ventoy/issues/815#issuecomment-819585071",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
***@***.***": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by vjr** [comment]
Hello @ventoy  I was trying to look at the source code but struggle a bit to make modification to load EFI in 64 KB aligned memory.

Since you are more familiar with codebase of ventoy and edk2 could you please try using the `AllocateAlignedPages()` method and pass the buffer and size params to `LoadImage()` function and provide a build of `Ventoy2Disk.exe` and I can test it and report back here?

**Comment by peracchi** [comment]
> This is just a workaround, I still haven't found the cause of the problem. However, I will add this workaround in the next release. But it is not enabled by default. So you need to press `ctrl+r` everytime you boot the two special ISO files.

Thanks for the "CTRL-R" trick, I was having the same problem with **neon-user-20211223-0945.iso**.

**Comment by tux-ware** [comment]
Hello @ventoy 

I just wanted to report that "CTRL+R" helped me too with KDE **neon-user-20211223-0945.iso** as well as with **linuxmint-20.x-*-64bit.iso** images. If I choose any of them in ventoy menu, without "CTRL+R", they show just a black screen with a small command prompt. If I press "CTRL+R" in ventoy menu and then press enter, the grub menu shows as expected, and I can continue with booting.

In the case of Linux Mint I could work around it, by pressing the enter key a second time, while seeing just a black screen, and then it booted. With KDE neon, pressing enter twice did not help, so I was glad to find the "CTRL+R" trick.

I tested it with secure boot off on a Dell XPS13 (9333) from 2014 (Haswell era laptop).
